examples/resources/awscc_amplify_domain/amplify_domain.tf

```diff
@@ -0,0 +1,35 @@
+resource "awscc_amplify_app" "example" {
+  name = "app"
+
+  # Setup redirect from https://example.com to https://www.example.com
+  custom_rules = [
+    {
+      source = "https://example.com"
+      status = "302"
+      target = "https://www.example.com"
+    },
+  ]
+}
+
+resource "awscc_amplify_branch" "main" {
+  app_id      = awscc_amplify_app.example.app_id
+  branch_name = "main"
+}
+
+resource "awscc_amplify_domain" "example" {
+  app_id      = awscc_amplify_app.example.app_id
+  domain_name = "example.com"
```

Comment:
Quick question, would it be any use case or example using Route53 CNAME creation?

Reply:
Hmm, potentially but may get in the weeds of having to explain CNAME validation. I pulled and refactored this example from the docs for the `aws` provider for this resource. I'm thinking they may have left it out for similar reasons: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/amplify_domain_association

Reply:
I could also potentially add something similar to this and then reference the domain name:
`resource "aws_route53_zone" "primary" {
  name = "example.com"
}`
but there are other resources that would also need to be added